web/src/engine/websites/Delitoon.ts

```
@@ -0,0 +1,174 @@
+            'x-balcony-timeZone': this.Timezone,
+            'X-platform': this.Platform,
+        };
+        if (this.Token) headers['authorization'] = ' Bearer ' + this.Token.accessToken.token;
+        return new Request(url, {
+            method: 'GET',
+            headers: headers
+        });
+    }
+
+    private CreatePostRequest(url: URL, body: string, contentType: string = 'application/json'): Request {
+        const headers: HeadersInit = {
+            Referer: this.URI.origin,
+            'x-balcony-id': this.BalconyID,
+            'x-balcony-timeZone': this.Timezone,
+            'X-platform': this.Platform,
+            'Content-type': contentType
+        };
+        if (this.Token) headers['authorization'] = ' Bearer ' + this.Token.accessToken.token;
+        return new Request(url, {
+            method: 'POST',
+            headers: headers,
+            body: body,
+        });
+    }
+
+    protected async GetToken() {
+
+        try {
+            //if token is undefined => try to fetch one
+            if (!this.Token) {
+                const url = new URL('/api/auth/session', this.URI);
+                const { user } = await FetchJSON<APIUser>(this.CreateRequest(url));
+                this.Token = user;
+                return;
+            }
+
+            //if token is defined, check for expiration  and refresh if needed
+            if (this.Token && Date.now() > this.Token.accessToken.expiredAt) { //24h expiration
+                const url = new URL('/api/balcony/auth/refresh', this.URI);
+                const body = JSON.stringify({
+                    accessToken: this.Token.accessToken.token,
+                    refreshToken: this.Token.refreshToken.token
+                });
+                const { result } = await FetchJSON<APIRefreshToken>(this.CreatePostRequest(url, body));
+                this.Token = result;
+            }
+        } catch (error) {
+            this.Token = undefined;
```

Shouldn't the exception be forwarded to be handled by the caller?
All errors such as network error, out of memory, ... are just silently ignored